setup.py

```diff
@@ -0,0 +1,134 @@
+            response.raise_for_status()
+            with open(file_path, 'w', encoding="utf-8") as f:
+                f.write(response.text)
+        except requests.exceptions.RequestException as e:
+            print(f"Error: Failed to download {file} from {file_url}.")
+            raise e
+
+
+def generate_protos():
+    """Generate python protobuf files from the proto files."""
+    from grpc_tools import protoc
+
+    protos_dir = f"{PROJECT_DIR}/_protos"
+    proto_files = PROTO_FILES
+
+    for proto_file in proto_files:
+        proto_path = os.path.join(protos_dir, proto_file)
+        if not os.path.exists(proto_path):
+            raise Exception(f"Error: {proto_file} not found.")
+        output_file = proto_path.replace('.proto', '_pb2.py')
+
+        if not os.path.exists(output_file) or os.path.getmtime(proto_path) > os.path.getmtime(output_file):
+            print(f"Compiling {proto_path}...")
+            command = [
+                'grpc_tools.protoc',
+                f'-I={protos_dir}',
+                f'--python_out={protos_dir}',
+                f'--grpc_python_out={protos_dir}',
+                proto_path
+            ]
+            if protoc.main(command) != 0:
+                raise Exception(f"Error: {proto_file} compilation failed")
+
+            # Fix the import path in the generated control_api_pb2
+            # https://github.com/protocolbuffers/protobuf/issues/1491#issuecomment-261914766
+            if "control_api" in proto_file:
+                with open(output_file, 'r') as file:
+                    filedata = file.read()
+                    newdata = filedata.replace(
+                        "import ank_base_pb2 as ank__base__pb2",
+                        "from . import ank_base_pb2 as ank__base__pb2")
+                with open(output_file, 'w') as file:
+                    file.write(newdata)
+
+
+setup(
+    description="Eclipse Ankaios Python SDK - provides a convenient Python interface for interacting with the Ankaios platform.",
+    long_description=open('README.md').read(),
+    long_description_content_type="text/markdown",
+    url="https://eclipse-ankaios.github.io/ankaios/latest/",
+    python_requires='>=3.6',
+    package_dir={'': '.'},
+    packages=find_packages(where="."),
+    include_package_data=True,
+    classifiers=[
+        "Programming Language :: Python :: 3",
+        "License :: OSI Approved :: Apache Software License",
+        "Operating System :: OS Independent",
+    ],
+    project_urls={
+        "Documentation": "https://eclipse-ankaios.github.io/ankaios/latest/",
+        "Source": "https://github.com/eclipse-ankaios/ank-sdk-python",
+        "Bug Tracker": "https://github.com/eclipse-ankaios/ank-sdk-python/issues",
+    },
+    install_requires=[
+        "protobuf==5.27.2",  # Protocol Buffers
+        "PyYAML",  # Used to parse manifest files
+    ],
+    setup_requires=[
+        "protobuf==5.27.2",  # Protocol Buffers
```

Comment:
The warning occurs because you have installed in your environment `protobuf` with version `5.28.1`. The proto files that are being generated have the `5.27.2` version, this is why in the `setup.py` I have this version being explicity installed.

Having another version will generate that warning (when importing ankaios_sdk) but nothing more, the functionality is ok. To fix this error, please install the SDK using

```python
cd path/to/sdk/ankaios_python_sdk
pip install .
```

This will install the SDK with the correct version for the `protobuf` dependency.